src/snowflake/connector/config_parser/__init__.py

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+from pathlib import Path
+from shutil import copyfile
+
+from tomlkit import parse
+
+from ..constants import config_file
+from .parser import ConfigParser
+
+# Create default configuration file, if it doesn't exist
+if not config_file.exists():
+    # Create default config file
+    default_config = Path(__file__).absolute().parent / "default_config.toml"
```

Comment:
just to double-check: This would create a default config file in the folder where connector is deployed? Is this a good idea? Package deployment may be in read-only, protected directories. If creating a default file is really necessary, why not create it in CWD and/or warn the user? I feel default_config files should either be created as part of setup.py during deployment or at most in CWD as it may be fair game to assume CWD is writable. 

Is the existence of this default config file required for later code to run?

Reply:
This line points to [default_config.py](https://github.com/snowflakedb/snowflake-connector-python/blob/mkeller/SNOW-765846-config_parser/src/snowflake/connector/config_parser/default_config.toml). We then copy this file to the user's [resolved config file](https://github.com/snowflakedb/snowflake-connector-python/blob/dfcc728f15fa2627c1f953651d80b94f8f2bcabb/src/snowflake/connector/constants.py#L33).

> I feel default_config files should either be created as part of setup.py during deployment or at most in CWD as it may be fair game to assume CWD is writable.

I don't think this is possible because of wheels, but I agree this would be better!

> Is the existence of this default config file required for later code to run?

It is not, Python connector will run happily without our new config file just like it has previously.

Reply:
Hm, so wheels won't allow to copy a default config file to some standard dir? Maybe there's a workaround for that, e.g. entry-point?

I may be a bit lost here understanding what points to what. Might be a good idea to put the logic here despite it's just a few lines into a function and add some comments. I find especially mentioning to which file the relative path to __file__ should point to helpful from your comment, as it may help track bugs down if there's some directory setup we couldn't think of yet.

Reply:
Good points.  I think what @sfc-gh-mkeller is saying is that installing wheels doesn't run any code.  It could contain non-code data files, but that probably doesn't work in a cross platform way.  Do we need `default_config.toml` to hang around after the `copyfile()` to `config_file` location completes?  If not, then just stick the default in a tempfile and let it get unlinked once the real config file is put in place.

Reply:
> I may be a bit lost here understanding what points to what. Might be a good idea to put the logic here despite it's just a few lines into a function and add some comments.

I'm also lost and don't really know what logic exactly you think I should put here. Could you take a stab at adding some comment here, maybe I could then expand on it!

> Hm, so wheels won't allow to copy a default config file to some standard dir?

Not as far as I know.

> Maybe there's a workaround for that, e.g. entry-point?

Not for libraries, for an application we could use those. This (import time) is essentially the entry-point for a library

Reply:
We synced up offline and I've dropped default file generation.